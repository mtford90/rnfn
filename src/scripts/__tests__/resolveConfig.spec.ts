import * as path from "path";
import { resolveConfig } from "../resolveConfig";

beforeEach(() => {
  jest.resetModules();
});

describe("config resolution", () => {
  test("override all", async () => {
    const p = path.resolve(__dirname, "./__mocks__/overrideAll.js");
    const config = await resolveConfig(p);
    expect(config).toMatchSnapshot();
  });

  test("override some", async () => {
    const p = path.resolve(__dirname, "./__mocks__/overrideSome.js");
    const config = await resolveConfig(p);
    expect(config).toMatchSnapshot();
  });

  describe("extending colors", () => {
    describe("with an extra colour", () => {
      it("should add the extra color", async () => {
        const p = path.resolve(__dirname, "./__mocks__/extendColors.js");
        const config = await resolveConfig(p);
        expect(config.theme.colors.mycolor).toEqual("pink");
      });

      it("should match the snapshot", async () => {
        const p = path.resolve(__dirname, "./__mocks__/extendColors.js");
        const config = await resolveConfig(p);
        expect(config).toMatchSnapshot();
      });
    });
  });

  describe("extending spacing", () => {
    describe("with extra spacing", () => {
      it("should add the extra spacing", async () => {
        const p = path.resolve(__dirname, "./__mocks__/extendSpacing.js");
        const config = await resolveConfig(p);
        expect(config.theme.spacing.lg).toEqual(32);
      });
    });
  });

  describe("extending fontSize", () => {
    describe("with extra fontSize", () => {
      it("should add the extra fontSize", async () => {
        const p = path.resolve(__dirname, "./__mocks__/extendFontSize.js");
        const config = await resolveConfig(p);
        expect(config.theme.fontSize.hello).toEqual([
          12,
          { letterSpacing: 32, lineHeight: 16 },
        ]);
      });
    });
  });

  describe("extending fontFamily", () => {
    it("should add the extra fontFamily", async () => {
      const p = path.resolve(__dirname, "./__mocks__/extendFontFamily.js");
      const config = await resolveConfig(p);
      expect(config.theme.fontFamily.customFont).toEqual({
        normal: {
          normal: { fontFamily: "myFont" },
          thin: { fontFamily: "myFont-thin" },
        },
      });
    });
  });
});
