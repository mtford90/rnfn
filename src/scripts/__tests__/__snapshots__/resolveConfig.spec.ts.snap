// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config resolution extending colors with an extra colour should match the snapshot 1`] = `
Object {
  "theme": Object {
    "colors": Object {
      "amber": Object {
        "100": "#fef3c7",
        "200": "#fde68a",
        "300": "#fcd34d",
        "400": "#fbbf24",
        "50": "#fffbeb",
        "500": "#f59e0b",
        "600": "#d97706",
        "700": "#b45309",
        "800": "#92400e",
        "900": "#78350f",
      },
      "black": "#000",
      "blue": Object {
        "100": "#dbeafe",
        "200": "#bfdbfe",
        "300": "#93c5fd",
        "400": "#60a5fa",
        "50": "#eff6ff",
        "500": "#3b82f6",
        "600": "#2563eb",
        "700": "#1d4ed8",
        "800": "#1e40af",
        "900": "#1e3a8a",
      },
      "blueGray": Object {
        "100": "#f1f5f9",
        "200": "#e2e8f0",
        "300": "#cbd5e1",
        "400": "#94a3b8",
        "50": "#f8fafc",
        "500": "#64748b",
        "600": "#475569",
        "700": "#334155",
        "800": "#1e293b",
        "900": "#0f172a",
      },
      "coolGray": Object {
        "100": "#f3f4f6",
        "200": "#e5e7eb",
        "300": "#d1d5db",
        "400": "#9ca3af",
        "50": "#f9fafb",
        "500": "#6b7280",
        "600": "#4b5563",
        "700": "#374151",
        "800": "#1f2937",
        "900": "#111827",
      },
      "cyan": Object {
        "100": "#cffafe",
        "200": "#a5f3fc",
        "300": "#67e8f9",
        "400": "#22d3ee",
        "50": "#ecfeff",
        "500": "#06b6d4",
        "600": "#0891b2",
        "700": "#0e7490",
        "800": "#155e75",
        "900": "#164e63",
      },
      "emerald": Object {
        "100": "#d1fae5",
        "200": "#a7f3d0",
        "300": "#6ee7b7",
        "400": "#34d399",
        "50": "#ecfdf5",
        "500": "#10b981",
        "600": "#059669",
        "700": "#047857",
        "800": "#065f46",
        "900": "#064e3b",
      },
      "fuchsia": Object {
        "100": "#fae8ff",
        "200": "#f5d0fe",
        "300": "#f0abfc",
        "400": "#e879f9",
        "50": "#fdf4ff",
        "500": "#d946ef",
        "600": "#c026d3",
        "700": "#a21caf",
        "800": "#86198f",
        "900": "#701a75",
      },
      "gray": Object {
        "100": "#f3f4f6",
        "200": "#e5e7eb",
        "300": "#d1d5db",
        "400": "#9ca3af",
        "50": "#f9fafb",
        "500": "#6b7280",
        "600": "#4b5563",
        "700": "#374151",
        "800": "#1f2937",
        "900": "#111827",
      },
      "green": Object {
        "100": "#dcfce7",
        "200": "#bbf7d0",
        "300": "#86efac",
        "400": "#4ade80",
        "50": "#f0fdf4",
        "500": "#22c55e",
        "600": "#16a34a",
        "700": "#15803d",
        "800": "#166534",
        "900": "#14532d",
      },
      "indigo": Object {
        "100": "#e0e7ff",
        "200": "#c7d2fe",
        "300": "#a5b4fc",
        "400": "#818cf8",
        "50": "#eef2ff",
        "500": "#6366f1",
        "600": "#4f46e5",
        "700": "#4338ca",
        "800": "#3730a3",
        "900": "#312e81",
      },
      "lightBlue": Object {
        "100": "#e0f2fe",
        "200": "#bae6fd",
        "300": "#7dd3fc",
        "400": "#38bdf8",
        "50": "#f0f9ff",
        "500": "#0ea5e9",
        "600": "#0284c7",
        "700": "#0369a1",
        "800": "#075985",
        "900": "#0c4a6e",
      },
      "lime": Object {
        "100": "#ecfccb",
        "200": "#d9f99d",
        "300": "#bef264",
        "400": "#a3e635",
        "50": "#f7fee7",
        "500": "#84cc16",
        "600": "#65a30d",
        "700": "#4d7c0f",
        "800": "#3f6212",
        "900": "#365314",
      },
      "mycolor": "pink",
      "orange": Object {
        "100": "#ffedd5",
        "200": "#fed7aa",
        "300": "#fdba74",
        "400": "#fb923c",
        "50": "#fff7ed",
        "500": "#f97316",
        "600": "#ea580c",
        "700": "#c2410c",
        "800": "#9a3412",
        "900": "#7c2d12",
      },
      "pink": Object {
        "100": "#fce7f3",
        "200": "#fbcfe8",
        "300": "#f9a8d4",
        "400": "#f472b6",
        "50": "#fdf2f8",
        "500": "#ec4899",
        "600": "#db2777",
        "700": "#be185d",
        "800": "#9d174d",
        "900": "#831843",
      },
      "purple": Object {
        "100": "#ede9fe",
        "200": "#ddd6fe",
        "300": "#c4b5fd",
        "400": "#a78bfa",
        "50": "#f5f3ff",
        "500": "#8b5cf6",
        "600": "#7c3aed",
        "700": "#6d28d9",
        "800": "#5b21b6",
        "900": "#4c1d95",
      },
      "red": Object {
        "100": "#fee2e2",
        "200": "#fecaca",
        "300": "#fca5a5",
        "400": "#f87171",
        "50": "#fef2f2",
        "500": "#ef4444",
        "600": "#dc2626",
        "700": "#b91c1c",
        "800": "#991b1b",
        "900": "#7f1d1d",
      },
      "rose": Object {
        "100": "#ffe4e6",
        "200": "#fecdd3",
        "300": "#fda4af",
        "400": "#fb7185",
        "50": "#fff1f2",
        "500": "#f43f5e",
        "600": "#e11d48",
        "700": "#be123c",
        "800": "#9f1239",
        "900": "#881337",
      },
      "teal": Object {
        "100": "#ccfbf1",
        "200": "#99f6e4",
        "300": "#5eead4",
        "400": "#2dd4bf",
        "50": "#f0fdfa",
        "500": "#14b8a6",
        "600": "#0d9488",
        "700": "#0f766e",
        "800": "#115e59",
        "900": "#134e4a",
      },
      "transparent": "transparent",
      "trueGray": Object {
        "100": "#f5f5f5",
        "200": "#e5e5e5",
        "300": "#d4d4d4",
        "400": "#a3a3a3",
        "50": "#fafafa",
        "500": "#737373",
        "600": "#525252",
        "700": "#404040",
        "800": "#262626",
        "900": "#171717",
      },
      "violet": Object {
        "100": "#ede9fe",
        "200": "#ddd6fe",
        "300": "#c4b5fd",
        "400": "#a78bfa",
        "50": "#f5f3ff",
        "500": "#8b5cf6",
        "600": "#7c3aed",
        "700": "#6d28d9",
        "800": "#5b21b6",
        "900": "#4c1d95",
      },
      "warmGray": Object {
        "100": "#f5f5f4",
        "200": "#e7e5e4",
        "300": "#d6d3d1",
        "400": "#a8a29e",
        "50": "#fafaf9",
        "500": "#78716c",
        "600": "#57534e",
        "700": "#44403c",
        "800": "#292524",
        "900": "#1c1917",
      },
      "white": "#fff",
      "yellow": Object {
        "100": "#fef3c7",
        "200": "#fde68a",
        "300": "#fcd34d",
        "400": "#fbbf24",
        "50": "#fffbeb",
        "500": "#f59e0b",
        "600": "#d97706",
        "700": "#b45309",
        "800": "#92400e",
        "900": "#78350f",
      },
    },
    "fontFamily": Object {},
    "fontSize": Object {
      "2xl": Array [
        24,
        Object {
          "lineHeight": 32,
        },
      ],
      "3xl": Array [
        30,
        Object {
          "lineHeight": 36,
        },
      ],
      "4xl": Array [
        36,
        Object {
          "lineHeight": 40,
        },
      ],
      "base": Array [
        16,
        Object {
          "lineHeight": 24,
        },
      ],
      "lg": Array [
        18,
        Object {
          "lineHeight": 28,
        },
      ],
      "sm": Array [
        14,
        Object {
          "lineHeight": 20,
        },
      ],
      "xl": Array [
        20,
        Object {
          "lineHeight": 28,
        },
      ],
      "xs": Array [
        12,
        Object {
          "lineHeight": 16,
        },
      ],
    },
    "opacity": Object {
      "0": "0",
      "100": "1",
      "25": ".25",
      "50": ".5",
      "75": ".75",
    },
    "spacing": Object {
      "0": 0,
      "0.5": 2,
      "1": 4,
      "1.5": 6,
      "10": 40,
      "11": 44,
      "12": 48,
      "14": 52,
      "16": 64,
      "2": 8,
      "2.5": 10,
      "20": 80,
      "24": 96,
      "28": 112,
      "3": 12,
      "3.5": 14,
      "32": 128,
      "36": 144,
      "4": 16,
      "40": 160,
      "44": 176,
      "48": 192,
      "5": 20,
      "52": 208,
      "56": 224,
      "6": 24,
      "60": 240,
      "64": 256,
      "7": 28,
      "72": 288,
      "8": 32,
      "80": 320,
      "9": 36,
      "96": 374,
      "dp": 1,
    },
  },
}
`;

exports[`config resolution override all 1`] = `
Object {
  "theme": Object {
    "colors": Object {
      "black": "#000",
      "white": "#fff",
    },
    "fontFamily": Object {},
    "fontSize": Object {
      "base": Array [
        16,
        Object {
          "lineHeight": 24,
        },
      ],
      "sm": Array [
        14,
        Object {
          "lineHeight": 20,
        },
      ],
      "xs": Array [
        12,
        Object {
          "lineHeight": 16,
        },
      ],
    },
    "opacity": Object {
      "0": "0",
      "100": "1",
      "25": ".25",
      "50": ".5",
      "75": ".75",
    },
    "spacing": Object {
      "lg": 20,
      "sm": 10,
      "xl": 30,
    },
  },
}
`;

exports[`config resolution override some 1`] = `
Object {
  "theme": Object {
    "colors": Object {
      "black": "#000",
      "white": "#fff",
    },
    "fontFamily": Object {},
    "fontSize": Object {
      "2xl": Array [
        24,
        Object {
          "lineHeight": 32,
        },
      ],
      "3xl": Array [
        30,
        Object {
          "lineHeight": 36,
        },
      ],
      "4xl": Array [
        36,
        Object {
          "lineHeight": 40,
        },
      ],
      "base": Array [
        16,
        Object {
          "lineHeight": 24,
        },
      ],
      "lg": Array [
        18,
        Object {
          "lineHeight": 28,
        },
      ],
      "sm": Array [
        14,
        Object {
          "lineHeight": 20,
        },
      ],
      "xl": Array [
        20,
        Object {
          "lineHeight": 28,
        },
      ],
      "xs": Array [
        12,
        Object {
          "lineHeight": 16,
        },
      ],
    },
    "opacity": Object {
      "0": "0",
      "100": "1",
      "25": ".25",
      "50": ".5",
      "75": ".75",
    },
    "spacing": Object {
      "0": 0,
      "0.5": 2,
      "1": 4,
      "1.5": 6,
      "10": 40,
      "11": 44,
      "12": 48,
      "14": 52,
      "16": 64,
      "2": 8,
      "2.5": 10,
      "20": 80,
      "24": 96,
      "28": 112,
      "3": 12,
      "3.5": 14,
      "32": 128,
      "36": 144,
      "4": 16,
      "40": 160,
      "44": 176,
      "48": 192,
      "5": 20,
      "52": 208,
      "56": 224,
      "6": 24,
      "60": 240,
      "64": 256,
      "7": 28,
      "72": 288,
      "8": 32,
      "80": 320,
      "9": 36,
      "96": 374,
      "dp": 1,
    },
  },
}
`;
